helpers/AsyncFromSyncIterator.js

```diff
@@ -0,0 +1,230 @@
+'use strict';
+
+var internalSlot = require('internal-slot');
+var isCallable = require('is-callable');
+
+var callBind = require('./callBind');
+var callBound = require('./callBound');
+var GetIntrinsic = require('../GetIntrinsic');
+var $isNaN = require('./isNaN');
+
+var $apply = GetIntrinsic('%Reflect.apply%', true) || callBound('%Function.prototype.apply%');
+var $asyncIterator = GetIntrinsic('%Symbol.asyncIterator%', true);
+var $toStringTag = GetIntrinsic('%Symbol.toStringTag%', true);
+var $TypeError = GetIntrinsic('%TypeError%');
+
+var undefined;
+
+// TODO: Use %AsyncIterator.from% once it's in ECMA-262
+
+/** @type {(o: object, p: string | symbol, Desc: import('es-abstract').PropertyDescriptor) => boolean} */
+var DefineOwnProperty = callBind(
+	require('./DefineOwnProperty'),
+	undefined,
+	function IsDataDescriptor(Desc) {
+		return !('[[Get]]' in Desc) && !('[[Set]]' in Desc);
+	},
+	GetIntrinsic('%Object.is%', true) || function SameValue(x, y) {
+		if (x === y) {
+			// 0 === -0, but they are not identical.
+			if (x === 0) {
+				return 1 / x === 1 / y;
+			}
+			return true;
+		}
+		return $isNaN(x) && $isNaN(y);
+	},
+	function FromPropertyDescriptor(Desc) {
+		var obj = {};
+		if ('[[Value]]' in Desc) {
+			obj.value = Desc['[[Value]]'];
+		}
+		if ('[[Writable]]' in Desc) {
+			obj.writable = Desc['[[Writable]]'];
+		}
+		if ('[[Get]]' in Desc) {
+			obj.get = Desc['[[Get]]'];
+		}
+		if ('[[Set]]' in Desc) {
+			obj.set = Desc['[[Set]]'];
+		}
+		if ('[[Enumerable]]' in Desc) {
+			obj.enumerable = Desc['[[Enumerable]]'];
+		}
+		if ('[[Configurable]]' in Desc) {
+			obj.configurable = Desc['[[Configurable]]'];
+		}
+		return obj;
+	}
```

Comment:
```suggestion
	FromPropertyDescriptor,
```
where FromPropertyDescriptor is `require('./FromPropertyDescriptor')` at module level